exercises/largest-series-product/largest_series_product.cr

```diff
@@ -0,0 +1,27 @@
+class Series
+  def initialize(series : String)
+    if series =~ /\D/
+      @series = [] of Int64
+    else
+      @series = series.chars.map(&.to_i).map(&.to_i64)
+    end
+  end
+
+  def largest_product(span : Int32) : Int64
+    return -1i64 if span > @series.size || (@series.empty? && span > 0) || span < 0
+    return 1i64 if span == 0 
+
+    product_from_sequence(create_sequences(span))
+  end
+
+  private def create_sequences(span : Int32) : Array(Array(Int64))
+    seq = @series.map_with_index do |n, i|
+      @series[i, span]
+    end
+    seq.select{|s|s.size == span}
```

Comment:
@mhelmetag @elorest I would suggest running `crystal tool format` on the whole project to automatically make the code follow the recommended style.

( Side note: this exercise can also be implemented in a single method chain: https://play.crystal-lang.org/#/r/14n2 )